task09-reactjs/my-app/src/components/CardGrid.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import CharacterCard from './cards/CharacterCard';
+import IslandCard from './cards/IslandCard';
+import MysticCard from './cards/MysticCard';
+
+const CardGrid = ({type, items}) => {
+
+  return (
+    <>
+      {items && items.length !== 0 ? (
+        type !== "character" ? (
+          type !== "island" ? (
+            <div className="mist-grid side-padding">
+              {items.map(item => { return (<MysticCard image={item.img} name={item.name} description={item.description}/>) })}
+            </div>
+            ) : (
+            <div className="card-grid side-padding" >
+              {items.map(item => { return (<IslandCard image={item.img} name={item.name} location={item.Location}/>) })}
+            </div>
+          )
+        ) : (
+          <div className="card-grid side-padding" >
+            {items.map(item => { return (<CharacterCard image={item.img} name={item.name}/>) })}
+          </div>
+        )
+      ) : (
+        null
+        //nothing to render
+      )}   
+    </>
```

Comment:
If you don't need to render something when no items are passed as params you could use
`items && items.length && (....)` instead, to avoid writing the null render.
Also to check the types of cards, you could a switch statement.